projects/js-packages/publicize-components/src/components/social-previews/constants.js

```diff
@@ -12,7 +12,37 @@ export const AVAILABLE_SERVICES = [
 		title: __( 'Facebook', 'jetpack' ),
 		icon: 'facebook',
 		name: 'facebook',
-		preview: props => <FacebookPreview type="article" { ...props } />,
+		preview: props => {
+			const { connection, title, excerpt, content, message, mediaDetails, attachedMediaDetails } =
+				props;
+			const attachedMediaData = attachedMediaDetails?.mediaData;
+			const { sourceUrl, width, height } = attachedMediaData || mediaDetails || {};
+
+			let user;
+			let imageMode;
+
+			if ( connection ) {
+				user = {
+					displayName: connection.display_name,
+					avatarUrl: connection.profile_picture,
+				};
+			}
+
+			if ( width && height && typeof width === 'number' && typeof height === 'number' ) {
+				imageMode = width > height ? 'landscape' : 'portrait';
+			}
+
+			return (
+				<FacebookPreview
+					{ ...props }
+					type="article"
+					user={ user }
+					customText={ message || excerpt || content || title }
+					customImage={ sourceUrl }
+					imageMode={ imageMode }
+				/>
+			);
+		},
```

Comment:
May be we can extract this to a separate component and keep the constants file cleaner?